### PR TITLE
chore(engine): migrate boundary event processor

### LINKED ISSUE #4494: Migrate Boundary Event to new BPMN lifecycle processor
**Description**

Refactor the current implementation of the BPMN element to match with the new BPMN lifecycle processors defined in #4333.

### PR DESCRIPTION
**Pull request by saig0**
## Description

* migrate the processor for boundary events

## Related issues

closes #4494 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: engine/src/main/java/io/zeebe/engine/nwe/event/BoundaryEventProcessor.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        .applyOutputMappings(context, element)
+        .ifRightOrLeft(
+            ok -> stateTransitionBehavior.transitionToCompleted(context),
+            failure -> incidentBehavior.createIncident(failure, context));
+  }
+
+  @Override
+  public void onCompleted(final ExecutableBoundaryEvent element, final BpmnElementContext context) {
+
+    stateTransitionBehavior.takeOutgoingSequenceFlows(element, context);
```
Comment (korthout) [inline engine/src/main/java/io/zeebe/engine/nwe/event/BoundaryEventProcessor.java:67]: Is the flownode that the boundary event is attached to already terminated when the boundary event completed? I would expect some logic here to check whether the boundary event is interrupting or not.

**Review by korthout — COMMENTED** [review]
LGTM, but I have 1 question

File: engine/src/main/java/io/zeebe/engine/nwe/event/BoundaryEventProcessor.java (hunk 1 above)
Comment (saig0) [reply engine/src/main/java/io/zeebe/engine/nwe/event/BoundaryEventProcessor.java:67]: This logic is in the activity itself. When the `activating` event is written for the boundary event then it was already triggered and the activity was terminated - if it is an interrupting boundary event. So, there is no much to do here.

**Review by korthout — APPROVED** [review]

**Pull request merged**